pages/hackerapps/[id]/basicinfo.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { getHackathonPaths, getHackathons } from '../../../utility/firebase'
+import Page from '../../../components/page'
+import { HACKER_APP_NAVBAR, COLOR } from '../../../constants'
+import QuestionCard from '../../../components/questionCard'
+import Icon from '../../../components/Icon'
+
+const HeaderContainer = styled.div`
+  display: flex;
+`
+
+const Header = styled.h1`
+  margin-top: 0;
+  font-weight: 600;
+  font-size: 32px;
+  color: ${COLOR.MIDNIGHT_PURPLE_DEEP};
+`
+
+const StyledQuestionButton = styled.button`
+  border: none;
+  background: transparent;
+  display: block;
+  margin: 10px auto;
+  &:hover {
+    transform: scale(1.2);
+  }
+  i {
+    transform: scale(1.3);
+  }
+`
+
+export default ({ id, hackathons }) => {
+  const [questions, setQuestions] = useState([
+    { title: '', description: '', type: '', options: [''], other: false, required: false },
+  ])
+
+  const addQuestion = () => {
+    setQuestions([...questions, { title: '', description: '', type: '', options: [''], other: false, required: false }])
+  }
```

Comment:
what are your thoughts on changing it so that a new question is inserted immediately after the question whose "add" button was clicked (maybe by passing an index to this function)? instead of adding it to the end

Reply:
you are so right